File: backend-spring-boot/src/test/java/no/vicx/backend/info/GitPropertiesControllerIntegrationTest.java
package no.vicx.backend.info;

import no.vicx.backend.testconfiguration.TestSecurityConfig;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;
import org.springframework.test.web.servlet.MockMvc;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

@SpringBootTest
@AutoConfigureMockMvc
@Import(TestSecurityConfig.class)
class GitPropertiesControllerIntegrationTest {

    @Autowired
    MockMvc mockMvc;

    @Test
    void getGitProperties() throws Exception {
        mockMvc.perform(get("/gitproperties"))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.branch").exists())
                .andExpect(jsonPath("$.commitId").exists())
                .andExpect(jsonPath("$.shortCommitId").exists())
                .andExpect(jsonPath("$.commitTime").exists());
    }
}

File: backend-spring-boot/src/test/kotlin/no/vicx/backend/info/GitPropertiesControllerIntegrationTest.kt
package no.vicx.backend.info

import io.kotest.core.spec.style.StringSpec
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.webmvc.test.autoconfigure.AutoConfigureMockMvc
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

@SpringBootTest
@AutoConfigureMockMvc
class GitPropertiesControllerIntegrationTest(
    mockMvc: MockMvc,
) : StringSpec({
        "when calling GET /gitproperties then expect result" {
            mockMvc
                .perform(get("/gitproperties"))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.branch").exists())
                .andExpect(jsonPath("$.commitId").exists())
                .andExpect(jsonPath("$.shortCommitId").exists())
                .andExpect(jsonPath("$.commitTime").exists())
        }
    })
